gnome-image-installer/util/gis-unattended-config.c

```diff
@@ -209,6 +220,45 @@ gis_unattended_config_populate_fields (GisUnattendedConfig *self,
           g_ptr_array_add (self->vendors, g_steal_pointer (&vendor));
           g_ptr_array_add (self->products, g_steal_pointer (&product));
         }
+      else if (g_str_has_prefix (*group, IMAGE_GROUP_PREFIX))
+        {
+          if (seen_image_group != NULL)
+            {
+              g_set_error (error, GIS_UNATTENDED_ERROR,
+                           GIS_UNATTENDED_ERROR_INVALID_IMAGE,
+                           /* Translators: this error refers to a configuration
+                            * file. The placeholders are all the name of
+                            * sections in an .ini-style file.
+                            */
+                           _("More than one %s section (%s and %s)"),
+                           "Image", seen_image_group, *group);
+              return FALSE;
+            }
+
+          seen_image_group = *group;
+          if (!key_file_get_optional_string (self->key_file,
+                                             *group, FILENAME_KEY,
+                                             &self->filename,
+                                             error) ||
+              !key_file_get_optional_string (self->key_file,
+                                             *group, BLOCKDEVICE_KEY,
+                                             &self->blockdevice,
+                                             error))
+            return FALSE;
+
+          if (self->blockdevice != NULL && *self->blockdevice == '\0')
```

Comment:
Might make sense to factor this validation out into a helper function.

Reply:
Done, and required `image` to be non-empty if specified at the same time.